File: src/leetcode/two-out-of-three/two-out-of-three.js
/**
 * Leetcode #2032
 * Difficulty: Easy
 * https://leetcode.com/problems/two-out-of-three/
 */
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @param {number[]} nums3
 * @return {number[]}
 */
export const twoOutOfThree = function (nums1, nums2, nums3) {
  const ans = new Set();

  const set1 = new Set();
  for (const num of nums1) {
    set1.add(num);
  }

  const set2 = new Set();
  for (const num of nums2) {
    set2.add(num);

    if (set1.has(num)) {
      ans.add(num);
    }
  }

  for (const num of nums3) {
    if (set1.has(num) || set2.has(num)) {
      ans.add(num);
    }
  }

  return [...ans];
};

File: src/leetcode/two-out-of-three/two-out-of-three.rs
impl Solution {
    pub fn two_out_of_three(nums1: Vec<i32>, nums2: Vec<i32>, nums3: Vec<i32>) -> Vec<i32> {
        let mut ans = std::collections::HashSet::new();
        let set1 = nums1.iter().collect::<std::collections::HashSet<_>>();
        let set2 = nums2.iter().collect::<std::collections::HashSet<_>>();

        for v in set2.clone() {
            if set1.contains(v) {
                ans.insert(v);
            }
        }

        for v in nums3.iter() {
            if set1.contains(v) || set2.contains(v) {
                ans.insert(v);
            }
        }

        ans.into_iter().cloned().collect()
    }
}